eth/downloader/downloader.go

```diff
@@ -818,6 +818,10 @@ func (d *Downloader) findAncestor(p *peerConnection, remoteHeader *types.Header)
 			return 0, errInvalidAncestor
 		}
 		p.log.Debug("Found common ancestor", "number", number, "hash", hash)
+		if number > localHeight {
```

Comment:
Our downloader has negotiated a common ancestor, but that ancestor is actually above what our blockchain considers it's head.